invoice/views.py

```diff
@@ -168,18 +185,62 @@ class InvoiceItemCreateView(CreateView):
 
 
 class VendorCreateView(CreateView):
-    """Create a new vendor."""
+    """Create a new vendor. Including a bank account and a new address."""
+    template_name = 'invoice/vendor_form.html'
     model = Vendor
-    fields = '__all__'
+    fields = ['name', 'company_name', 'tax_id']
     success_url = reverse_lazy('vendor-list')
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if self.request.POST:
+            context['address_form'] = AddressForm(self.request.POST)
+            context['bank_form'] = BankAccountForm(self.request.POST)
+        else:
+            context['address_form'] = AddressForm()
+            context['bank_form'] = BankAccountForm()
+        return context
+
+    def form_valid(self, form):
+        """Create a new vendor, a new address and a new bank account."""
+        address_form = AddressForm(self.request.POST)
+        address = address_form.save()
+        bank_account_form = BankAccountForm(self.request.POST)
+        bank_account = bank_account_form.save()
+        vendor = form.save(commit=False)
+        vendor.address = address
+        vendor.bank_account = bank_account
+        return super().form_valid(form)
+
 
 class VendorUpdateView(UpdateView):
-    """Update an existing vendor."""
+    """Update an existing vendor. Including the bank account and address."""
+    template_name = 'invoice/vendor_form.html'
     model = Vendor
-    fields = '__all__'
+    fields = ['name', 'company_name', 'tax_id']
     success_url = reverse_lazy('vendor-list')
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if self.request.POST:
+            context['address_form'] = AddressForm(self.request.POST)
+            context['bank_form'] = BankAccountForm(self.request.POST)
+        else:
+            context['address_form'] = AddressForm(instance=self.object.address)
+            context['bank_form'] = BankAccountForm(instance=self.object.bank_account)
+        return context
+
+    def form_valid(self, form):
+        """Updates an existing vendor including the address and the bank account."""
+        address_form = AddressForm(instance=self.object.address, data=self.request.POST)
+        if address_form.is_valid():
+            address_form.save()
+        bank_account_form = BankAccountForm(instance=self.object.bank_account,
+                                            data=self.request.POST)
+        if bank_account_form.is_valid():
```

Comment:
what about error handling?
what happens when the address_form/bank_account is invalid?